vector-db-proxy/src/adaptors/pinecone/apis.rs

```diff
@@ -72,32 +64,32 @@ impl VectorDatabase for PineconeClient {
         &self,
         collection_create: CollectionCreate,
     ) -> Result<VectorDatabaseStatus, VectorDatabaseError> {
-        let region = collection_create.region.unwrap_or_default();
-        // Namespace is set the datasource ID if it's a manged Vector DB
-        //
-        let _namespace = collection_create
-            .clone()
-            .namespace
-            .map_or(collection_create.clone().collection_name, |n| n);
-        // The index is set to the region if it's a manged vector database
-        // If it's BYO the index is the collection_name in the datasource document from mongo
         let index_name = collection_create
```

Comment:
Once again you have removed the concept of a namespace. I don't know how this could possibly work

Reply:
Ideally the user would always use the same index when creating a new datasource but if he uses a new index name then we have no choice but to create a  new index. There is no api in pinecone to create a new namespace. It automatically creates a new namespace when we upsert a doc to a non existing namespace